File: front/src/models/dtos/reference/createReferenceDto.ts
import { Address } from '@/models/entities/address';
import { Company } from '@/models/entities/company';

/**
 * DTO for creating a new reference.
 */
export type CreateReferenceDto = {
  /**
   * The first name of the referer.
   */
  firstName: string;

  /**
   * The last name of the referer.
   */
  lastName: string;

  /**
   * The address of the referer.
   */
  address: Address;

  /**
   * The email address of the referer.
   */
  email: string;

  /**
   * The phone number of the referer.
   */
  phone: string;

  /**
   * The company where the referer works.
   */
  company: Omit<Company, 'id'>;
};

File: backend/api_gateway/src/main/kotlin/com/linkedout/backend/dto/reference/CreateReferenceDto.kt
package com.linkedout.backend.dto.reference

import com.linkedout.backend.model.Address

data class CreateReferenceDto(
    val firstName: String,
    val lastName: String,
    val address: Address,
    val phone: String,
    val email: String,
    val company: Company
) {
    data class Company(
        val name: String
    )
}
